### PR TITLE
Validate required fields

### PR DESCRIPTION
**Pull request by j0sv**
## PT: https://www.pivotaltracker.com/story/show/151126573 ## 

- Add required to input fields required before user submit an order. 
- Defect: https://www.pivotaltracker.com/story/show/151122486

### CODE REVIEW

**Review by tochman — CHANGES_REQUESTED** [review]
In the model, please use 'validates_presence_of' validation. Just setting the fields to required does not cut it.

**Comment by j0sv** [comment]
The address and other information don't exist then the instance is created. The validation needs to be made in the controller if it should be made on server side also.

File: app/models/order.rb [hunk 1]
```diff
@@ -1,10 +1,22 @@
 class Order < ApplicationRecord
   acts_as_shopping_cart_using :order_item
+  validates :delivery_date,       presence: true, :if => :submitted?
```
Comment (tochman) [inline app/models/order.rb]: Old hash syntax 😜

**Review by tochman — CHANGES_REQUESTED** [review]
Please add specs for the validations

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ def create
       dish = Dish.find(dish_id)
       if dish_value.to_i > 0
         if @order.add(dish, dish.price, dish_value.to_i)
-          flash[:notice] = "#{dish.name} was successfully added to order!"
+          #flash[:notice] = "#{dish.name} was successfully added to order!"
```
Comment (tochman) [inline app/controllers/orders_controller.rb:17]: If you remove this line then the entire if statement makes no sense. Or does it? Perhaps use a modifier:
```
flash[:alert] = 'Item not added, try again!' unless @order.add(dish, dish.price, dish_value.to_i)
```
but I'm not sure if I like it...

File: app/controllers/orders_controller.rb [hunk 3]
```diff
@@ -29,8 +29,12 @@ def destroy
 
   def update
     @order = get_order
-    @order.update(order_params)
-    redirect_to confirm_order_path
+    if @order.update(order_params)
+      redirect_to confirm_order_path
+    else
+      flash[:alert] = 'Error then saving order!'
```
Comment (tochman) [inline app/controllers/orders_controller.rb]: Error WHEN saving order. Do we have a test for this `else`block?  Is there a use case`

**Review by tochman — CHANGES_REQUESTED** [review]
I still don't see any specs for the validations and the new methods on the Order model.

**Comment by tochman** [comment]
Merge conflicts need to be resolved and unit specs added. Not mergeable in this state.

**Review by tochman — CHANGES_REQUESTED** [review]
There are comments in the code. Please remove them. Can't see the relevant CSS classes for the flash. Am I missing something? I will pull down the branch to take a closer look at the unit specs. Can't see why you are updating the subject before testing the validations.

**Comment by j0sv** [comment]
There aren't any css classes. Removing the class from the HTML element.

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**